client/state/user-settings/reducer.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { merge, omit } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	USER_SETTINGS_FETCH,
+	USER_SETTINGS_FETCH_SUCCESS,
+	USER_SETTINGS_FETCH_FAILURE,
+	USER_SETTINGS_SAVE_SUCCESS,
+	USER_SETTINGS_RECEIVE,
+	USER_SETTINGS_UNSAVED_SET,
+	USER_SETTINGS_UNSAVED_REMOVE,
+} from 'state/action-types';
+import { createReducer } from 'state/utils';
+import { decodeEntities } from 'lib/formatting';
+
+/*
+ * Decodes entities in those specific user settings properties
+ * that the REST API returns already HTML-encoded
+ */
+function decodeUserSettingsEntities( data ) {
+	const decodedValues = {
+		display_name: data.display_name && decodeEntities( data.display_name ),
+		description: data.description && decodeEntities( data.description ),
+		user_URL: data.user_URL && decodeEntities( data.user_URL )
+	};
+
+	return merge( {}, data, decodedValues );
+}
+
+/**
+ * Returns the updated remote values state after an action has been dispatched.
+ * The remote values state reflects preferences which are persisted to the REST
+ * API current user settings endpoint.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const settings = createReducer( false, {
+	[ USER_SETTINGS_RECEIVE ]: ( state, { values } ) => {
+		return merge( {}, state || {}, decodeUserSettingsEntities( values ) );
+	}
+} );
```

Comment:
> You mentioned that there is overlap in the settings here as with other reducers? Are we duplicating the same state here where instead we could reuse the other state (a superset of the settings we need here)?

Actually, it's the other way around: There is a `client/state/preferences` module that fetches the `me/settings` endpoint and stores a *subset* of the settings, namely the `calypso_preferences` key. The value of that key is a JSON structure that is stored in `state.preferences`.

The `client/state/preferences` module could be merged into this one after it's finished.